services/management/commands/update_vantaa_parking_areas.py

```diff
@@ -57,78 +103,100 @@ def get_multi_geom(self, obj):
-
         municipality = Municipality.objects.get(name_fi="Vantaa")
-        type = AdministrativeDivisionType.objects.get(type="parking_area")
-
-        service = restapi.FeatureService(SRC_SERVICE_URL)
-        parking_areas = service.layer(SRC_LAYER_NAME)
-        features = parking_areas.query()
 
-        logger.info("Found {} parking areas for Vantaa".format(features.count))
-        logger.info("Importing parking areas...")
+        for data_source in DATA_SOURCES:
+            start_time = time()
+            num_parking_areas_updated = 0
 
-        updated_parking_areas = []
-        for feature in features:
-            geom = self.get_multi_geom(
-                GEOSGeometry(str(feature.get("geometry")), srid=SRC_SRID)
-            )
-            geom.transform(src_to_dest)
-            props = feature.get("properties")
-            origin_id = str(props.get("objectid")) if props.get("objectid") else None
-            if not origin_id:
-                logger.warning("Parking area has no origin ID, skipping")
-                continue
-            name_fi = props.get("tyyppi")
-            defaults = {
-                "name_fi": name_fi,
-                "municipality": municipality,
-                "type": type,
-                "extra": props,
-                "origin_id": origin_id,
-            }
             try:
-                name_en = PARKING_NAME_TRANSLATIONS[name_fi]["en"]
-                name_sv = PARKING_NAME_TRANSLATIONS[name_fi]["sv"]
-                defaults["name_en"] = name_en
-                defaults["name_sv"] = name_sv
-            except KeyError:
-                logger.warning(f"No translation for {name_fi}")
-
-            division, _ = AdministrativeDivision.objects.update_or_create(
-                ocd_id=OCD_ID_VANTAA_PARKING_BASE + origin_id,
-                defaults=defaults,
+                division_type = AdministrativeDivisionType.objects.get(
+                    type=data_source["type"]
+                )
+            except AdministrativeDivisionType.DoesNotExist:
+                division_name = (
+                    data_source["layer_name"].replace("MUOKATTAVA", "").strip()
+                )
+                division_type = AdministrativeDivisionType.objects.create(
+                    type=data_source["type"], name=division_name
+                )
+
+            service = restapi.FeatureService(data_source["service_url"])
+            parking_areas = service.layer(data_source["layer_name"])
+            features = parking_areas.query()
+
+            readable_name = data_source["type"].replace("_", " ").strip() + "s"
+            logger.info(f"Found {features.count} {readable_name} for Vantaa")
+            logger.info(f"Importing {readable_name}...")
+
+            updated_parking_areas = []
+            for feature in features:
+                geom = self.get_multi_geom(
+                    GEOSGeometry(str(feature.get("geometry")), srid=SRC_SRID)
```

Comment:
Should there be a None-check for `feature.get("geometry")`?

Reply:
This is part of old code (just moved) but yeah, I could of course add it.

Reply:
Added the skip and a test for it.